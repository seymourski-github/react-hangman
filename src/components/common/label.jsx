import React from 'react';
import PropTypes from 'prop-types';

const mapProps = ({
  children, className, disabled, hidden, htmlFor, text
}) => ({
  children: text||children,
  className, disabled, hidden, htmlFor
});

const mapStyle = ({ alignSelf, style, flexItem, visible }) => ({
  ...style,
  alignSelf,
  flex: flexItem,
  visibility: visible ? 'visible':'hidden'
});

export const Label = props => {
  return (
    <label {...mapProps(props)}
      style={mapStyle(props)}
    />
  );
};

Label.propTypes = {
  alignSelf: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  flexItem: PropTypes.string,
  hidden: PropTypes.bool,
  style:  PropTypes.object,
  visible: PropTypes.bool,
  text: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
};

Label.defaultProps = {
  alignSelf: '',
  className: 'label',
  disabled: false,
  hidden: false,
  flex: '',
  style:  null,
  text: '',
  visible: true
}
