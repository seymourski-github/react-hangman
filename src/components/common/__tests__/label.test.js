import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Label } from '../label';

describe('<Label /> Component', function() {
  const component = renderer.create(<Label text="My label" />);
  const tree = component.toJSON();

  it("renders and matches our snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});
