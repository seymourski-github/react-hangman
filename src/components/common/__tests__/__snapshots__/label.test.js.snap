// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Component renders and matches our snapshot 1`] = `
<label
  className="label"
  disabled={false}
  hidden={false}
  htmlFor={undefined}
  style={
    Object {
      "alignSelf": "",
      "flex": undefined,
      "visibility": "visible",
    }
  }
>
  My label
</label>
`;
